ui/src/fields/ColoredMark/ColoredMark.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { Popover } from 'antd'
+import { useSelector } from 'react-redux'
+import { AppState } from '../../interfaces/storeSlices'
+import { ColoredMarkField } from '../../interfaces/widget'
+import { DataItem } from '@tesler-ui/core/interfaces/data'
+import styles from './ColoredMark.module.css'
+
+export interface ColoredMarkProps {
+    value: string
+    cursor: string
+    widgetName: string
+    meta: ColoredMarkField
+}
+
+function ColoredMark({ value, cursor, widgetName, meta }: ColoredMarkProps) {
+    const bcName = useSelector((state: AppState) => state.view.widgets?.find(i => i.name === widgetName)?.bcName)
+    const data = useSelector((state: AppState) => bcName && state.data[bcName]?.find(item => item.id === cursor)) as DataItem
+
+    if (!value) {
+        return null
+    }
+
+    const displayedDescription = meta?.hintKey ? data?.[meta.hintKey] : null
+
+    if (!displayedDescription) {
```

Comment:
I think `!value` and `!displayedDescription` can be combined in one check, they literally one line next to each other

Reply:
exactly